kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/discovery/informer/IndexerCompositeFactory.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.discovery.informer;
+
+import io.kubernetes.client.common.KubernetesListObject;
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.informer.SharedIndexInformer;
+import io.kubernetes.client.informer.cache.Indexer;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.kubernetes.client.informer.SharedIndexInformerFactory;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+
+/**
+ * Factory for {@link IndexerComposite}.
+ *
+ * @author Pavol Gressa
+ * @since 3.2
+ */
+@Internal
+@Singleton
+public class IndexerCompositeFactory {
```

Comment:
This factory generates informers of given type for provided set of namespaces and returns the IndexerComposite for the resource lookups.